### ISSUE TITLE
bug: e2e tests failing when updating to 4.20.0

### ISSUE THREAD

**Issue by danyball** [ionic-team/stencil#6072]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.20.0

### Current Behavior

We have around 700 e2e tests running with 4.19.2. After updating to 4.20.0 tests fails on our server (jenkins and gitlab ci tested). On my local Mac I can not see any errors.

- some of them "App did not load in allowed time. Please ensure the content loads a stencil application."
- some with random errors. 
- the test script just stopped at a point - no more logging / gitlab stops those jobs after 1h

### Expected Behavior

tests still running

### System Info

```shell
Server:
- tested with several node versions, from 18 to LTS
- debian bookworm OS
- HeadlessChrome/121.0.6167.85 (with puppeteer 21.11.0 - also tested 21.0.0)
- on gitlab ci:
   - `- apt-get update && apt-get install -y libnss3 libdbus-1-3 libatk1.0-0 libx11-xcb1 libxcomposite1 libxcursor1 libxdamage1 libxi6 libxtst6 libxrandr2 libasound2 libatk-bridge2.0-0 libgtk-3-0 libcups2 libpango1.0-0 libxss1 fonts-liberation xdg-utils
`

local:
System: node 18.20.5 (LTS also working)
    Platform: darwin (24.1.0)
   CPU Model: Apple M1 Max (10 cpus)
    Compiler: /node_modules/@stencil/core/compiler/stencil.js
       Build: 1732216626
     Stencil: 4.22.3 🐤
  TypeScript: 5.5.4
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.31.1
```

### Steps to Reproduce

- having many tests (do not find a reproducer)
- use specific stencil version
- run e2e tests on a server env

### Code Reproduction URL

sorry no

### Additional Information

https://discord.com/channels/520266681499779082/1304108899657318442/1304108899657318442

**Comment by christian-bromann** [comment]
Hey @danyball 👋 when loading a Stencil component using the Stencil testrunner in an e2e test case we verify that all components resolved their `componentOnReady` hook. If this has not happened before the `config.testing.browserWaitUntil` timeout is reached it fails with given error.

At this point there is not much I can do as this is very difficult to reproduce. If you have any ideas what may can cause this I would be curious to know.

**Comment by danyball** [comment]
Thanks Christian. Did you change something runtime timing relevant in 4.20?
Maybe something in 

runtime: hydrate shadow dom first (https://github.com/ionic-team/stencil/issues/5911) ([ccf1a89](https://github.com/ionic-team/stencil/commit/ccf1a8941f732cb53d57785ecbe03388e744a1cd))
runtime: make isSameVnode return false on initial render in a hydration case (https://github.com/ionic-team/stencil/issues/5891) ([82a7bb9](https://github.com/ionic-team/stencil/commit/82a7bb9ead3dc637b646db09b6687a8d0c2735a2))
?

Maybe the new if in "isSameVnode" could go wrong on slow machines or so?!

**Comment by christian-bromann** [comment]
@danyball potentially but I am not certain.

**Comment by tishoyanchev** [comment]
We are facing the same or a similar issue. @christian-bromann, @danyball 
Two days ago the e2e unit tests started failing the checks on Github actions. When I run them locally, all of them fail with the error "_`App did not load in allowed time. Please ensure the content loads a stencil application`_."
On Github Actions, the log is showing this error:
```
Puppeteer old Headless deprecation warning:
    In the near future `headless: true` will default to the new Headless mode
    for Chrome instead of the old Headless implementation. For more
    information, please see https://developer.chrome.com/articles/new-headless/.
    Consider opting in early by passing `headless: "new"` to `puppeteer.launch()`
    If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.
[16:09.3]  transpile started ...
[ ERROR ]  Failed to launch the browser process!
           [0115/151609.612447:FATAL:zygote_host_impl_linux.cc(1[27](https://github.com/Infineon/infineon-design-system-stencil/actions/runs/12791252232/job/35658932635#step:5:28))] No usable
           sandbox! Update your kernel or see
           https://chromium.googlesource.com/chromium/src/+/main/docs/linux/suid_sandbox_development.md
           for more information on developing with the SUID sandbox. If you
           want to live dangerously and need an immediate workaround, you can
           try using --no-sandbox. TROUBLESHOOTING:
           https://pptr.dev/troubleshooting Error: Failed to launch the browser
           process! [0115/151609.612447:FATAL:zygote_host_impl_linux.cc(127)]
           No usable sandbox! Update your kernel or see
           https://chromium.googlesource.com/chromium/src/+/main/docs/linux/suid_sandbox_development.md
           for more information on developing with the SUID sandbox.
```

My stencil version is 4.23.0
My puppeteer version is 21.9.0

Any ideas what to do?

**Comment by tishoyanchev** [comment]
> [@tishoyanchev](https://github.com/tishoyanchev) we are seeing the same issues in our pipeline and investigating the issue.

@christian-bromann 
Hi, I saw that you guys released a new version for Stencil that is meant to fix this problem, however, I still have it. 
Here's a more detailed explanation of the problem I'm facing:

- Some months ago our dependabot created a PR for puppeteer's new version. We tested it, and found out that it's not compatible with Stencil, and the test runners were failing with the same error described in this Issue. So we decided to keep using their old version `^21.9.0` until they fix their problem. 
- Few days ago, the e2e unit tests check on GitHub actions started failing. When I tested it locally, all components failed with the above described error. 
- Yesterday I tried removing the `^` symbol on puppeteer because I suspected a newer version could have been applied. I just installed the fixed `21.9.0` version, and surprisingly all tests passed locally! but, they still failed on github actions, although this time, only 9 were failing, while the rest were passing. I cleaned cache, but it didn't make any difference.
- Today I updated stencil to your latest version, and tried with puppeteer version `^21.9.0`, `21.9.0` and latest `24.1.0`, but 9 tests keep failing locally and on github actions. The error is the same: _`App did not load within 10000ms. Please ensure the content loads a stencil application.`_

Any ideas what to do? 
Is it working for you? With what version?

**Comment by christian-bromann** [comment]
@tishoyanchev can anyone provide a reproduction case?

**Comment by johnjenkins** [comment]
To those affected, I believe there is a conflation of a couple of issues here.

1) The initially reported one of either `App did not load in allowed time` or `Please ensure the content loads a stencil application`

This was addressed via https://github.com/ionic-team/stencil/pull/6098 / v4.24.0

2) `App did not load within 10000ms` - A secondary issue (introduced by https://github.com/ionic-team/stencil/pull/6098) which changed the default timeout for all e2e jest tests from 30s to 10s 

The reverting of that change is merged and awaiting release. If you can't wait, you can set the `timeout` on the `newE2EPage({... timeout: 30000 })`

**Comment by tishoyanchev** [comment]
> To those affected, I believe there is a conflation of a couple of issues here.
> 
> 1. The initially reported (either `App did not load in allowed time` or `Please ensure the content loads a stencil application`)
> 
> This was addressed via [#6098](https://github.com/ionic-team/stencil/pull/6098) / v4.24.0
> 
> 2. `App did not load within 10000ms` - A secondary issue (introduced by [fix(testing): update puppeteer, default to 'new' headless #6098](https://github.com/ionic-team/stencil/pull/6098)) which changed the default timeout for all e2e tests from 30s to 10s
> 
> The reverting of that change is merged and awaiting release. If you can't wait, you can set the `timeout` on the `newE2EPage({... timeout: 30000 })`

Do you guys have any time frame for the 2nd issue?

**Comment by danyball** [comment]
Thanks for trying to help me out with Bug no.1. 

Upgraded to stencil 4.24 and puppeteer 24.1:
When starting all the tests about 10% of them are failing. But when picking out one failing suite and just run this one file alone, all tests are green. 
I can also see Bug no. 2 in the logs, and maybe the other failing tests in that suite are follow ups. Tried adding the timeout:30000 to all newE2EPage() - but there still happens error "App did not load within 10000ms."

**Comment by christian-bromann** [comment]
> Do you guys have any time frame for the 2nd issue?

No, any contributions are appreciated. Do you have a minimal reproduction case?

Just fyi: we had to update Puppeteer as the version we were running wasn't compatible with GitHub Actions. It was running a very old Chrome version which would crash.

**Comment by danyball** [comment]
Dont have a reproducer yet. 
Trying to run 100 tests in stencil-component-starter project.
Copy pasted one of our components to that project and added it to the tests.
But everything works.

@tishoyanchev :
1. Maybe it just will fail when Node is trying to run too many tests in parallel?! 
2. Or one of the tests includes a complex component, having some more logic when initializing?!
3. How many tests do you have in your project?
4. Are the test suite works when you just run this single one?

**Comment by tishoyanchev** [comment]
> To those affected, I believe there is a conflation of a couple of issues here.
> 
> 1. The initially reported (either `App did not load in allowed time` or `Please ensure the content loads a stencil application`)
> 
> This was addressed via [#6098](https://github.com/ionic-team/stencil/pull/6098) / v4.24.0
> 
> 2. `App did not load within 10000ms` - A secondary issue (introduced by [fix(testing): update puppeteer, default to 'new' headless #6098](https://github.com/ionic-team/stencil/pull/6098)) which changed the default timeout for all e2e tests from 30s to 10s
> 
> The reverting of that change is merged and awaiting release. If you can't wait, you can set the `timeout` on the `newE2EPage({... timeout: 30000 })`

@johnjenkins, @christian-bromann 
Hi, 
Just wanted to let you know that I just tried out your latest version (4.25.1), and now only 5 tests fail with the same error but it's not saying that it didn't load within 10000 like before, instead it's saying it didn't load within 30000ms, which I think is what you wanted to revert back to. 
So, do you have any idea why those 5 tests are failing?

**Comment by danyball** [comment]
Bug no. 2 in this thread is a duplicate of https://github.com/ionic-team/stencil/issues/4782 in my opinion. Something wrong with running tests in parallel with the new headless chrome.
The workaround `--runInBand` **fixes my failing tests**, but is not practical because of the time all the tests needs.

**Comment by danyball** [comment]
@tishoyanchev Could you try the workaround form the other ticket? Is that fixing your failing tests? Please post your insights to https://github.com/ionic-team/stencil/issues/4782 as this ticket also has a reproducer.

I will close this ticket. **Solution**:
- use Stencil 4.25.0
- use puppeteer 24.1.0
- run tests with `runInBand` workaround
- use `browserExecutablePath: executablePath()` in stencil.config

**Comment by tishoyanchev** [comment]
> [@tishoyanchev](https://github.com/tishoyanchev) Could you try the workaround form the other ticket? Is that fixing your failing tests? Please post your insights to [#4782](https://github.com/ionic-team/stencil/issues/4782) as this ticket also has a reproducer.
> 
> I will close this ticket. **Solution**:
> 
> * use Stencil 4.25.0
> * use puppeteer 24.1.0
> * run tests with `runInBand` workaround
> * use `browserExecutablePath: executablePath()` in stencil.config

Hi, I am trying to test your workaround right now, but stupid question, where do I put the `runInBand`?

**Comment by christian-bromann** [comment]
> where do I put the `runInBand`?

This is not well documented unfortunately but you can pass it as CLI argument, e.g. `npx stencil test --runInBand --e2e`

**Comment by danyball** [comment]
Opened up a new ticket for bug no. 2 (in this thread): https://github.com/ionic-team/stencil/issues/6157 Lets continue there @tishoyanchev

**Comment by christian-bromann** [comment]
Let's continue conversation in #6157
